controllers/configurationpolicy_controller.go

```diff
@@ -1011,6 +1012,33 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 
 				// Otherwise, set the resolved data for use in further processing
 				plc.Spec.ObjectTemplates[i].ObjectDefinition.Raw = resolvedTemplate.ResolvedJSON
+			} else if isRawObjTemplate {
+				// marshal raw template into YAML if that has not already been done by the template resolution fn
+				var yamlObj interface{}
+
+				err = yaml.Unmarshal(rawData, &yamlObj)
+				if err != nil {
+					addTemplateErrorViolation("Error unmarshaling YAML for raw template", err.Error())
+
+					return
+				}
+
+				resolvedJSON, err := json.Marshal(yamlObj)
```

Comment:
Optional: I think you can use `sigs.k8s.io/yaml` instead to avoid the JSON Marshal and Unmarshal code. I assume you are doing this because we don't have `yaml` tags on the struct fields, and `sigs.k8s.io/yaml` handles that for you I believe.